packages/cli/src/config/PublishDetails.ts

```diff
@@ -343,6 +345,35 @@ const getAndroidDetails = async (
   }
 };
 
+const checkAbis = async (apkPath: string) => {
+  try {
+    const { stdout } = await runExec(`zipinfo -s ${apkPath} | grep \.so$`);
+    const amV7libs = [...stdout.matchAll(/lib\/armeabi-v7a\/(.*)/g)].flatMap(permission => permission[1]);
+    const x86libs = [...stdout.matchAll(/lib\/x86\/(.*)/g)].flatMap(permission => permission[1]);
+    const x8664libs = [...stdout.matchAll(/lib\/x86_64\/(.*)/g)].flatMap(permission => permission[1]);
+    if (amV7libs.length > 0 || x86libs.length > 0 || x8664libs.length > 0) {
+
+      const messages = [
+        `Solana dApp Store only supports arm64-v8a abi.`,
+        `Your apk file contains following unsupported abis`,
+        ... amV7libs.length > 0 ? [`\narmeabi-v7a:\n` + amV7libs] : [],
+        ... x86libs.length > 0 ? [`\nx86:\n` + x86libs] : [],
+        ... x8664libs.length > 0 ? [`\nx86_64:\n` + x8664libs] : [],
+        `\n\nAlthough your app might work, these library files are unused and increase the size of apk file making the user experience worse while installing/updating your app.`,
```

Comment:
"might" will likely worry developers who don't necessarily understand ABIs. I'd rephrase this to emphasize that this isn't a functionality issue, but rather a usability issue causing longer download times for users.

Reply:
Updated.